### PR TITLE
Fix column width calculations

### PR DESCRIPTION
**Pull request by bkonyi**
We weren't taking the expansion caret into account, causing the largest node to be slightly cut off.

Fixes https://github.com/flutter/devtools/issues/4786

### CODE REVIEW

**Comment by bkonyi** [comment]
It looks like the test I added when I made this change doesn't actually catch overflows since we have `TextOverflow.ellipsis` set. Do you have any suggestions on how to modify that test?

**Comment by kenzieschmoll** [comment]
> Do you have any suggestions on how to modify that test?

You could find all the text widgets and verify none of them contain the substring '...'

File: packages/devtools_app/lib/src/shared/table/column_widths.dart [hunk 1]
```diff
@@ -121,17 +121,22 @@ extension TreeColumnWidthExtension<T extends TreeNode<T>>
...
           if (width > longestWidth) {
             longestWidth = width;
           }
         }
-        width = longestWidth;
+        // Add two characters to account for expansion caret and ensure we
+        // always display the full column title.
+        width = max(
+              max(longestWidth, column.minWidthPx ?? 0),
+              column.title.length,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/table/column_widths.dart:137]: should this be `column.title.length *  characterWidth`?

File: packages/devtools_app/test/shared/table_test.dart [hunk 2]
```diff
@@ -1006,7 +1006,34 @@ void main() {
           ),
         ),
       );
+
       expect(find.byWidget(table), findsOneWidget);
+      expect(
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/shared/table_test.dart:1011]: just to make sure, does this part of the test you added fail without the fix you added in this PR? Might be good to add a comment "Regression test for (<github issue link>)" to make that clear

File: packages/devtools_app/lib/src/shared/table/column_widths.dart [hunk 3]
```diff
@@ -121,17 +121,22 @@ extension TreeColumnWidthExtension<T extends TreeNode<T>>
...
           }
         }
-        width = longestWidth;
+        // Add two characters to account for expansion caret and ensure we
+        // always display the full column title.
+        width = max(
+              max(longestWidth, column.minWidthPx ?? 0),
+              column.title.length * characterWidth,
+            ) +
+            characterWidth * 2;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/table/column_widths.dart:139]: do we need to estimate here or do we know the exact size of the expansion caret?

**Review by kenzieschmoll — APPROVED** [review]
a couple more comments, but overall lgtm

File: packages/devtools_app/lib/src/shared/table/column_widths.dart (hunk 3 above)
Comment (bkonyi) [reply packages/devtools_app/lib/src/shared/table/column_widths.dart:139]: This is the exact size (one character for `^` and one for the whitespace).

**Pull request merged**